volatility3/framework/plugins/linux/mountinfo.py

```diff
@@ -0,0 +1,227 @@
+                    return None
+
+                # Global root?
+                if mnt.vol.offset != parent.vol.offset:
+                    dentry = mnt.get_mnt_mountpoint()
+                    mnt = parent
+                    vfsmnt = mnt.mnt
+                    continue
+
+                return None
+
+            parent = dentry.d_parent
+            dname = dentry.d_name.name_as_str()
+            path_reversed.append(dname.strip("/"))
+            dentry = parent
+
+        path = "/" + "/".join(reversed(path_reversed))
+        return path
+
+    @classmethod
+    def get_mountinfo(cls, mnt, task) -> Union[None, Tuple[int, int, str, str, str, List[str], 
+                                                           List[str], str, str, List[str]]]:
+        """Extract various information about a mount point.
+        It mimics the Linux kernel show_mountinfo function.
+        """
+        mnt_root = mnt.get_mnt_root()
+        if not mnt_root:
+            return None
+
+        mnt_root_path = mnt_root.path()
+        superblock = mnt.get_mnt_sb()
+
+        mnt_id: int = mnt.mnt_id
+        parent_id: int = mnt.mnt_parent.mnt_id
+
+        st_dev = f"{superblock.major}:{superblock.minor}"
+
+        path_root = cls._do_get_path(mnt, task.fs.root)
+        if path_root is None:
+            return None
+
+        mnt_opts: List[str] = []
+        mnt_opts.append(mnt.get_flags_access())
+        mnt_opts.extend(mnt.get_flags_opts())
+
+        # Tagged fields
+        fields: List[str] = []
+        if mnt.is_shared():
+            fields.append(f"shared:{mnt.mnt_group_id}")
+
+        if mnt.is_slave():
+            master = mnt.mnt_master.mnt_group_id
+            fields.append(f"master:{master}")
+            dominating_id = mnt.get_dominating_id(task.fs.root)
+            if dominating_id and dominating_id != master:
+                fields.append(f"propagate_from:{dominating_id}")
+
+        if mnt.is_unbindable():
+            fields.append("unbindable")
+
+        mnt_type = superblock.get_type()
+
+        devname = mnt.get_devname()
+        if not devname:
+            devname = "none"
+
+        sb_opts: List[str] = []
+        sb_opts.append(superblock.get_flags_access())
+        sb_opts.extend(superblock.get_flags_opts())
+
+        return MountInfoData(mnt_id, parent_id, st_dev, mnt_root_path, path_root, mnt_opts, fields, 
+                             mnt_type, devname, sb_opts)
+
+    def _get_mnt_namespace_mountpoints(self, mnt_namespace):
+        mnt_type = self._get_symbol_fullname("mount")
+        if not self.context.symbol_space.has_type(mnt_type):
+            # Old kernels ~ 2.6
+            mnt_type = self._get_symbol_fullname("vfsmount")
+
+        for mount in mnt_namespace.list.to_list(mnt_type, "mnt_list"):
+            yield mount
+            
+    def _get_tasks_mountpoints(self, pids: Iterable[int]):
+        self._vmlinux = self.context.modules[self.config['kernel']]
```

Comment:
`__init__` should call super and also define any local state (like `_vmlinux`).

Reply:
I'd also be happy if the symbol_tablename were instead passed manually into `_get_mnt_namespace_mountpoints` and `_get_symbol_fullname`.